test/Spago/Build.purs

```diff
@@ -99,65 +101,212 @@ spec = Spec.around withTempDir do
+            spago [ "build" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency.txt")
+
+        Spec.describe "appear in the test package and" do
 
-      Spec.describe "warns about unused dependencies when" do
-        let
-          setupSrcUnusedDeps spago = do
-            spago [ "init", "--name", "7368613235362d2f444a2b4f56375435646a59726b53586548" ] >>= shouldBeSuccess
-            FS.writeTextFile (Path.concat [ "src", "Main.purs" ]) "module Main where\nimport Prelude\nmain = unit"
-            -- get rid of "Compiling ..." messages and other compiler warnings
-            spago [ "build" ] >>= shouldBeSuccess
+          let
+            setupTestTransitiveTests spago = do
+              spago [ "init", "--name", "7368613235362d34312f4e59746b7869335477336d33414d72" ] >>= shouldBeSuccess
+              spago [ "install", "maybe" ] >>= shouldBeSuccess
+              FS.writeTextFile (Path.concat [ "src", "Main.purs" ]) $ Array.intercalate "\n"
+                [ "module Main where"
+                , "import Prelude"
+                , "import Data.Maybe as Maybe"
+                , "import Effect as Effect"
+                , "import Effect.Console as Console"
+                , "main = unit"
+                ]
+              FS.writeTextFile (Path.concat [ "test", "Test", "Main.purs" ]) $ Array.intercalate "\n"
+                [ "module Test.Main where"
+                , "import Prelude"
+                , "import Data.Maybe as Maybe"
+                , "import Effect as Effect"
+                , "import Effect.Console as Console"
+                , "import Control.Alt as Alt"
+                , "main = unit"
+                ]
+              -- get rid of "Compiling ..." messages and other compiler warnings
+              spago [ "build" ] >>= shouldBeSuccess
+
+          Spec.it "passed --pedantic-packages CLI flag" \{ spago, fixture } -> do
+            setupTestTransitiveTests spago
+            spago [ "build", "--pedantic-packages" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency-test.txt")
+
+          Spec.it "package config has 'pedantic_packages: true'" \{ spago, fixture } -> do
+            setupTestTransitiveTests spago
+            addPedanticPackagesToTest
+            spago [ "build" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency-test.txt")
+
+      Spec.describe "fails with warnings about unused dependencies" do
+
+        Spec.describe "in a source package when" do
+
+          let
+            setupSrcUnusedDeps spago installConsoleAndEffect = do
+              spago [ "init", "--name", "7368613235362d2f444a2b4f56375435646a59726b53586548" ] >>= shouldBeSuccess
+              when installConsoleAndEffect do
+                spago [ "install", "--test-deps", "console", "effect" ] >>= shouldBeSuccess
+              FS.writeTextFile (Path.concat [ "src", "Main.purs" ]) "module Main where\nimport Prelude\nmain = unit"
```

Comment:
I think we could do with a `writeMain :: Array String -> Effect Unit` helper (maybe a `writeTestMain` too?)